src/ocrmypdf/ocr_engine/rapidocr.py

```diff
@@ -0,0 +1,440 @@
+
+log = logging.getLogger(__name__)
+
+# Make the debug messages more visible
+log.info("Initializing RapidOCR engine module")
+
+# Try to import RapidOCR and track its availability
+RAPIDOCR_AVAILABLE = False
+IMPORT_ERROR_MESSAGE = ""
+
+try:
+    # Import with verbose logging to diagnose any issues
+    log.info("Attempting to import RapidOCR...")
+    from rapidocr_onnxruntime import RapidOCR
+
+    # Get version info
+    rapidocr_version = "unknown"
+    try:
+        # Try to get version directly
+        from rapidocr_onnxruntime import __version__ as rapidocr_version
+    except ImportError:
+        # Try to get version from the package metadata
+        try:
+            import pkg_resources
+
+            rapidocr_version = pkg_resources.get_distribution(
+                "rapidocr_onnxruntime"
+            ).version
+        except (pkg_resources.DistributionNotFound, ImportError):
+            # Try to get version from the module object
+            try:
+                rapidocr_version = getattr(RapidOCR, "__version__", "unknown")
+            except (AttributeError, TypeError):
+                log.warning("RapidOCR version not available")
+
+    # Test if RapidOCR can actually be initialized
+    try:
+        test_engine = RapidOCR(use_angle_cls=False)
+        RAPIDOCR_AVAILABLE = True
+        log.info(
+            f"✓ RapidOCR successfully imported and initialized, version: {rapidocr_version}"
+        )
+    except Exception as e:
+        IMPORT_ERROR_MESSAGE = f"RapidOCR initialization failed: {str(e)}"
+        log.warning(f"✗ {IMPORT_ERROR_MESSAGE}")
+
+except ImportError as e:
+    IMPORT_ERROR_MESSAGE = str(e)
+    rapidocr_version = 'not installed'
+    log.warning(f"✗ RapidOCR import failed: {IMPORT_ERROR_MESSAGE}")
+
+
+class RapidOcrEngine(OcrEngine):
+    """Implements OCR with RapidOCR."""
+
+    def __init__(self):
+        """Initialize the RapidOCR engine."""
+        if not RAPIDOCR_AVAILABLE:
+            log.warning("RapidOCR engine initialized but the library is not available")
+        else:
+            log.info("RapidOCR engine initialized successfully")
+
+    @staticmethod
+    def version() -> str:
+        """Return the version of RapidOCR."""
+        if not RAPIDOCR_AVAILABLE:
+            return "not installed"
+        return rapidocr_version
+
+    @staticmethod
+    def creator_tag(options) -> str:
+        """Return a creator tag for PDF metadata."""
+        return f"RapidOCR {RapidOcrEngine.version()}"
+
+    def __str__(self):
+        return f"RapidOCR {self.version()}"
+
+    @staticmethod
+    def languages(options) -> Set[str]:
+        """Return the languages supported by RapidOCR."""
+        # RapidOCR primarily supports Chinese, English and other languages
+        # but does not have the same language-specific models as Tesseract
+        # Return a set of common languages it works well with
+        return {'chi_sim', 'chi_tra', 'eng'}
```

Comment:
Is there a way to obtain a list of all installed language models from the RapidOCR API?